### ISSUE TITLE
image upload problem

### ISSUE THREAD

**Issue by saimayaqub** [pokeb/asi-http-request#244]
Hello,

I want to upload a  UIImage object to a REST url using POST method. This is how I am trying to achieve it:

```
    NSURL *postURL = [NSURL URLWithString:urlString];
NSData *imageData = UIImageJPEGRepresentation(imageViewPhotoTaken.image, 0.2);
ASIFormDataRequest *request = [ASIFormDataRequest requestWithURL:postURL];
[request appendPostData:imageData];
[request setRequestMethod:@"POST"];
[request setDelegate:self];
[request setDidFinishSelector:@selector(requestDone:)];
[request setDidFailSelector:@selector(requestWentWrong:)];
[request startAsynchronous];
```

NSLog(@"Response Data ==== %@ , Response Status Code ==== %@", [request responseData], [request responseStatusCode]);

How may i get to know if my image was posted well or not? Both responseData and responseStatusCode return me the null values.  Kindly help me out. I am stuck here since long.
Looking for the response.

Thanks in advance.

**Comment by pokeb** [comment]
Are you logging just after [request startAsynchronous], rather than in your requestDone: method? If so, you can't do this  because you're using an asynchronous request. Async requests return control immediately, so you have to wait for requestDone: or requestWentWrong: to be called to find out what happened.

Otherwise, is requestWentWrong being called?

**Comment by saimayaqub** [comment]
My problem is fixed now.  I had not mentioned the content type in my request header.
After adding this line of code, 
[request addRequestHeader:@"Content-Type" value:@"image/jpeg"];

I was able to post the image successfully.
Also, you are right, that i was logging right after [request startAsynchronous] call. As soon as i changed it to [request startSynchronous] , it started giving me proper status code values in responseData. I also made use of the other options like reading the responseData in selectors like requestDone. Your code really helped me a lot.

Thank you very much.
